codalab/rest/bundles.py

```diff
@@ -253,7 +259,9 @@ def _create_bundles():
         created_uuids.append(bundle_uuid)
         bundle_class = get_bundle_subclass(bundle['bundle_type'])
         bundle['owner_id'] = request.user.user_id
-        if issubclass(bundle_class, UploadedBundle) or query_get_bool('wait_for_upload', False):
+        if final_state:
+            bundle['state'] = final_state
+        elif issubclass(bundle_class, UploadedBundle) or query_get_bool('wait_for_upload', False):
             bundle['state'] = State.UPLOADING
```

Comment:
From your PR description, would the bundle without any content hit this line and be stuck in uploading?

Reply:
That's the original issue we are trying to fix. Basically, for the future, when a user `wadd` bundle without any content, the logic will first hit `final_state` check and then set the bundle state to be whatever the final state is and skip setting it to be `UPLOADING` and never gets it changed to final states.